janusgraph-core/src/main/java/org/janusgraph/graphdb/database/management/ManagementSystem.java

```diff
@@ -247,6 +248,9 @@ public synchronized void commit() {
             managementLogger.sendCacheEviction(updatedTypes, evictGraphFromCache, updatedTypeTriggers, getOpenInstancesInternal());
             for (JanusGraphSchemaVertex schemaVertex : updatedTypes) {
                 schemaCache.expireSchemaElement(schemaVertex.longId());
+                for (JanusGraphTransaction tx : graph.getOpenTransactions()) {
+                    tx.expireSchemaElement(schemaVertex.longId());
```

Comment:
I didn't verify but I think this only solves the problem partially. The same NPE could occur if the transaction runs on a different JanusGraph instance (typically on another machine).

Reply:
I added the code to ManagementLogger::read as well

Reply:
Thanks! I am just thinking if we could test this. Probably try opening two graphs in a single test? I am not sure if we have such tests in the code base.

Reply:
I looked for tests to copy but didn't find any.  Let me know if you know of one.

@li-boxuan , otherwise, are you happy with this PR?  Thanks

Reply:
No problem, I can add a test case for that afterward. Thank you!